### PR TITLE
[Workspace] Add ACL related test cases

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description

This PR introduces ACL related test cases for workspace.

### Issues Resolved

[List any issues this PR will resolve]

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_acl.spec.js [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+let ownerWorkspaceId = '';
+let datasourceId = '';
+
+const getPolicy = (permission, userName) => ({
+  [permission]: {
+    users: [userName],
+  },
+});
+
+const NONE_DASHBOARDS_ADMIN_USERNAME = 'workspace-acl-test';
```
Comment (raintygao) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_acl.spec.js:29]: Nit: should be NON?

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_acl.spec.js [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        );
+        setupWorkspace(readOnlyWorkspaceName, datasourceId).then(
+          (value) => (readOnlyWorkspaceId = value)
+        );
+        setupWorkspace(libraryWriteWorkspaceName, datasourceId).then(
+          (value) => (libraryWriteWorkspaceId = value)
+        );
+        setupWorkspace(ownerWorkspaceName, datasourceId).then(
+          (value) => (ownerWorkspaceId = value)
+        );
```
Comment (raintygao) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_acl.spec.js:113]: Nit: Should them be put together into a function?

**Review by raintygao — APPROVED** [review]

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_acl.spec.js (hunk 1 above)
Comment (SuZhou-Joe) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_acl.spec.js]: Sure, updated.

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_acl.spec.js (hunk 2 above)
Comment (SuZhou-Joe) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_acl.spec.js]: Yeah, done for that.

**Review by ruanyl — APPROVED** [review]

**Pull request merged**